lib/CL/devices/tbb/tbb.cc

```diff
@@ -0,0 +1,393 @@
+  POCL_MEM_FREE (name);
+#else
+  snprintf (res, 1000, "tbb-%s", HOST_DEVICE_BUILD_HASH);
+#endif
+  return res;
+}
+
+unsigned int
+pocl_tbb_probe (struct pocl_device_ops *ops)
+{
+  int env_count = pocl_device_get_env_count(ops->device_name);
+  /* Env was not specified, default behavior was to use 1 tbb device */
+  if (env_count < 0)
+    return 1;
+
+  return env_count;
+}
+
+static cl_device_partition_property tbb_partition_properties[2]
+    = { CL_DEVICE_PARTITION_EQUALLY, CL_DEVICE_PARTITION_BY_COUNTS };
+
+char scheduler_initialised = 0;
+
+cl_int
+pocl_tbb_init (unsigned j, cl_device_id device, const char* parameters)
+{
+  struct data *d;
+  cl_int ret = CL_SUCCESS;
+  int err;
+
+  d = (struct data *) calloc (1, sizeof (struct data));
+  if (d == NULL)
+    return CL_OUT_OF_HOST_MEMORY;
+
+  d->current_kernel = NULL;
+  device->data = d;
+
+  pocl_init_default_device_infos (device);
+  device->extensions = HOST_DEVICE_EXTENSIONS;
+
+  device->on_host_queue_props
+      = CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_PROFILING_ENABLE;
+
+  /* hwloc probes OpenCL device info at its initialization in case
+     the OpenCL extension is enabled. This causes to printout 
+     an unimplemented property error because hwloc is used to
+     initialize global_mem_size which it is not yet. Just put 
+     a nonzero there for now. */
+  device->global_mem_size = 1;
+  err = pocl_topology_detect_device_info (device);
+  if (err)
+    ret = CL_INVALID_DEVICE;
+
+  /* device->max_compute_units was set up by topology_detect,
+     but we use the TBB library (result should be the same).
+     task_area initialization is optional and max_concurrency
+     can be retrieved without prior initialization. */
+  tbb::task_arena ta;
+  //ta.initialize();
+  device->max_compute_units = ta.max_concurrency();
+
+  pocl_cpuinfo_detect_device_info(device);
+  pocl_set_buffer_image_limits(device);
+
+  /* in case hwloc doesn't provide a PCI ID, let's generate
+     a vendor id that hopefully is unique across vendors. */
+  const char *magic = "pocl";
+  if (device->vendor_id == 0)
+    device->vendor_id =
+      magic[0] | magic[1] << 8 | magic[2] << 16 | magic[3] << 24;
+
+  device->vendor_id += j;
+
+  /* tbb device has no partitioning support
+  device->max_sub_devices = device->max_compute_units;
+  device->num_partition_properties = 2;
+  device->partition_properties = tbb_partition_properties;
+  device->num_partition_types = 0;
+  device->partition_type = NULL;*/
+
+  if (!scheduler_initialised)
+    {
+      scheduler_initialised = 1;
+      pocl_init_dlhandle_cache();
+      pocl_init_kernel_run_command_manager();
+      tbb_scheduler_init (device);
+    }
+  /* system mem as global memory */
+  device->global_mem_id = 0;
+  return ret;
+}
+
+cl_int
+pocl_tbb_uninit (unsigned j, cl_device_id device)
+{
+  struct data *d = (struct data*)device->data;
+
+  if (scheduler_initialised)
+    {
+      tbb_scheduler_uninit ();
```

Comment:
maybe search for " (" or is this the coding style here? doesn't seem to be consistent on calls

Reply:
pocl has an inconsistent coding style for this and I don't feel like the one to make a decision/fix it.